include/zephyr/drivers/bbram/it8xxx2.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024 Google Inc
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_ZEPHYR_DRIVERS_BBRAM_IT8XXX2_H_
+#define INCLUDE_ZEPHYR_DRIVERS_BBRAM_IT8XXX2_H_
+
+#include <stdint.h>
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/emul_bbram.h>
+
+/** Device config */
+struct bbram_it8xxx2_config {
+	/** BBRAM base address */
+	uintptr_t base_addr;
+	/** BBRAM size (Unit:bytes) */
+	int size;
+};
+
+int it8xxx2_emul_backend_set_data(const struct emul *target, size_t offset, size_t count,
+				  const uint8_t *buffer);
+
+int it8xxx2_emul_backend_get_data(const struct emul *target, size_t offset, size_t count,
+				  uint8_t *buffer);
```

Comment:
These routines don't need to be public if you move the EMUL_DT_INST_DEFINE call into the emulator file.

Reply:
The emulator declaration sets the same config pointer as the driver. So I needed to declare the device and emulator in the same file. I originally had all of the logic in the same file and these functions as static, @alevkoy requested that I separate the logic which does make it cleaner with this one little exception.

Reply:
If this data needs to be shared between the driver and emulator, then you should create a private header file under the drivers/bbram directory.  The include/zephyr directory should only be used for public APIs.

Reply:
Done